test/e2e/fqdn_dns_cache_test.go

```diff
@@ -0,0 +1,350 @@
+	require.NoError(t, err, "Error when creating custom DNS service: %v", err)
+
+	setDnsServerAddressInAntrea(t, data, customDnsService.Spec.ClusterIP)
+
+	podCount := 2
+	agnHostPodIps := make([]*PodIPs, podCount)
+	for i := 0; i < podCount; i++ {
+		agnHostPodIps[i] = createHttpAgnhostPod(t, data, randName(agnHostPodNamePreFix), map[string]string{agnHostLabelKey: agnHostLabelValue})
+	}
+
+	ipv4, ipv6 := extractIPs(agnHostPodIps[0])
+
+	dnsConfigData := createDnsConfig(t, ipv4, ipv6)
+	customDnsConfigMapObject, err := data.CreateConfigMap(data.testNamespace, customDnsConfigName, dnsConfigData, nil, false)
+	require.NoError(t, err, "failed to create custom dns ConfigMap: %v", err)
+
+	createDnsPod(t, data)
+
+	k8sUtils, err = NewKubernetesUtils(data)
+	require.NoError(t, err, "error getting k8s utils %+v", err)
+
+	_ = buildFqdnPolicy(t, data)
+
+	createToolBoxPod(t, data, customDnsService.Spec.ClusterIP)
+
+	curlFqdn := func(podName, containerName, fqdn string) error {
+		cmd := []string{"curl", fqdn}
+		stdout, stderr, err := data.RunCommandFromPod(data.testNamespace, podName, containerName, cmd)
+		if err != nil {
+			return fmt.Errorf("error when running command '%s' on Pod '%s': %v, stdout: <%v>, stderr: <%v>",
+				strings.Join(cmd, " "), podName, err, stdout, stderr)
+		}
+		return nil
+	}
+
+	assert.Eventually(t, func() bool {
+		t.Logf("trying to curl the fqdn %v", testFullyQualifiedDomainName)
+		err = curlFqdn(toolBoxPodName, toolboxContainerName, testFullyQualifiedDomainName)
+		return assert.NoError(t, err)
+	}, 2*time.Second, 100*time.Millisecond)
+
+	// DIG to get actual IP , to be sure.
+	digResponse, err := k8sUtils.runDNSQuery(toolBoxPodName, toolboxContainerName, data.testNamespace, testFullyQualifiedDomainName, false, customDnsService.Spec.ClusterIP)
+	require.NoError(t, err, "failed to get IP of FQDN using DIG from toolbox pod : %v", err)
+	fqdnIp := digResponse.String()
+
+	t.Logf("received ip using dig for test fqdn %+v ", fqdnIp)
+
+	ipv4, ipv6 = extractIPs(agnHostPodIps[1])
+	if ipv6 != "" {
+		t.Logf("New IPs to update to DNS | ipv4 : %v, ipv6 : %+v", ipv4, ipv6)
+	} else {
+		t.Logf("New IPs to update to DNS | ipv4 : %v", ipv4)
+	}
+
+	// Update the custom DNS configMap
+	UpdatedCustomDNSconfig := createDnsConfig(t, ipv4, ipv6)
+
+	customDnsConfigMapObject.Data = UpdatedCustomDNSconfig
+	err = data.UpdateConfigMap(customDnsConfigMapObject)
+	require.NoError(t, err, "failed to update configmap with new IP : %v", err)
+	if ipv6 != "" {
+		t.Logf("successfully updated dns configMap with new IPs | ipv4 : %+v, ipv6 :%+v", ipv4, ipv6)
+	} else {
+		t.Logf("successfully updated dns configMap with new IPs | ipv4 : %+v", ipv4)
+	}
+
+	require.NoError(t, data.setPodAnnotationToRandomValue(data.testNamespace, customDnsPodName, randomPatchAnnotationKey), "failed to update custom dns pod annotation.")
+
+	defer setDnsServerAddressInAntrea(t, data, "")
```

Comment:
This `defer` statement should be located right after the first call to `setDnsServerAddressInAntrea`